File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/Floater.java
package edu.uchicago.gerber._08final.mvc.model;

import java.awt.*;
import java.util.ArrayList;
import java.util.List;

public abstract class Floater extends Sprite {

    public Floater() {

        setTeam(Team.FLOATER);

        setExpiry(250);
        setRadius(50);


        //set random DeltaX
        setDeltaX(somePosNegValue(10));

        //set random DeltaY
        setDeltaY(somePosNegValue(10));

        //set random spin
        setSpin(somePosNegValue(10));

        //cartesian points which define the shape of the polygon
        List<Point> listPoints = new ArrayList<>();
        listPoints.add(new Point(5, 5));
        listPoints.add(new Point(4,0));
        listPoints.add(new Point(5, -5));
        listPoints.add(new Point(0,-4));
        listPoints.add(new Point(-5, -5));
        listPoints.add(new Point(-4,0));
        listPoints.add(new Point(-5, 5));
        listPoints.add(new Point(0,4));

        setCartesians(CommandCenter.pointsListToArray(listPoints));

    }

    @Override
    public void draw(Graphics g) {
        renderVector(g);
    }

}

File: src/main/java/edu/uchicago/gerber/_08final/mvc/model/Floater.kt
package edu.uchicago.gerber._08final.mvc.model

import edu.uchicago.gerber._08final.mvc.controller.Utils.pointsListToArray
import edu.uchicago.gerber._08final.mvc.model.Movable.Team
import java.awt.Color
import java.awt.Graphics
import java.awt.Point

abstract class Floater : Sprite() {
    init {
        team = Team.FLOATER
        //default values, all of which can be overridden in the extending concrete classes
        expiry = 250
        color = Color.WHITE
        radius = 50
        //set random DeltaX
        deltaX = somePosNegValue(10).toDouble()
        //set random DeltaY
        deltaY = somePosNegValue(10).toDouble()
        //set random spin
        spin = somePosNegValue(10)

        //cartesian points which define the shape of the polygon
        val listPoints: MutableList<Point> = ArrayList()
        listPoints.add(Point(5, 5))
        listPoints.add(Point(4, 0))
        listPoints.add(Point(5, -5))
        listPoints.add(Point(0, -4))
        listPoints.add(Point(-5, -5))
        listPoints.add(Point(-4, 0))
        listPoints.add(Point(-5, 5))
        listPoints.add(Point(0, 4))
        cartesians = pointsListToArray(listPoints)
    }

    override fun draw(g: Graphics) {
        renderVector(g)
    }
}
